### ISSUE TITLE
[Usage]: How to use logits processors with max_num_seqs > 1?

### ISSUE THREAD

**Issue by alejopaullier96** [vllm-project/vllm#13553]
### Your current environment

Say you have a simple `LLM` class with `max_num_seqs`>1:
```python
llm = LLM(
    model_name,
    max_num_seqs=2
)
```
And a simple `LogitsProcessor`, for example, the following which forces the LLM to generate a certain phrase at the end of the generation:
```python
class ForceLastPhraseLogitsProcessor:
    """
    A logits processor which forces LLMs to use the given phrase before they finalize their answers.
    Most common use cases can be providing references, thanking user with context etc.

    Parameters
    ----------
    phrase (str): The phrase to be generated by LLM before the end of its speech.
    tokenizer (PreTrainedTokenizer): The tokenizer used by the LLM.
    """
    def __init__(self, phrase: str, tokenizer: PreTrainedTokenizer):
        self.eos_token_id = tokenizer.eos_token_id
        self.phrase_tokens = tokenizer.encode(phrase, add_special_tokens=False)
        self.index = 0

    def __call__(self, prompt_tokens_ids: List[int], past_token_ids: List[int], scores: torch.Tensor) -> torch.Tensor:
        if scores.argmax() == self.eos_token_id and self.index == 0:
            scores[self.phrase_tokens[self.index]] = scores.max() + 1
            self.index += 1
        elif len(self.phrase_tokens) > self.index > 0:
            scores[self.phrase_tokens[self.index]] = scores.max() + 1
            self.index += 1

        return scores
```

### Problem

Since in this example we have `max_num_seqs`>1 the generation is paralellized and breaks the internal logic of the logits processor as, for instance, process 1 gets `self.index = 1, 3, 5, 7, ...` and process 2 gets `self.index = 0, 2, 4, 6, ...`.

Methods that involve having some sort of counters do not work when having more than 1 generation process.

Is there an alternative to handle these cases?

### How would you like to use vllm

I want to know how a logits processor can handle more than 1 generation and if there is any additional parameter that can be passed to the `__call__` method other than the one shown in the example.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
